projects/plugins/boost/app/modules/cache/Boost_File_Cache.php

```diff
@@ -0,0 +1,123 @@
+		 */
+		for ( $i = 0; $i < 10; $i += 2 ) {
+			$path .= substr( $key, $i, 2 ) . '/';
+		}
+
+		return $path;
+	}
+
+	/*
+	 * Returns the cache filename for the given request, or current request.
+	 *
+	 * @param array $args - an array containing the request_uri, cookies array, and get array representing the request.
+	 * @return string - The cache path + filename for the given request.
+	 */
+	private function cache_filename( $args = array() ) {
+		$defaults = array(
+			'request_uri' => $this->request_uri,
+			'cookies'     => $_COOKIE, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			'get'         => $_GET, // phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.NonceVerification.Recommended
+		);
+		$args     = array_merge( $defaults, $args );
+
+		return $this->path( $args['request_uri'] ) . $this->cache_key( $args ) . '.html';
+	}
+
+	/*
+	 * Outputs the cached page if it exists for the given request, or current request.
+	 *
+	 * @param array $args - an array containing the request_uri, cookies array, and get array representing the request.
+	 * @return bool - false if page was not cached.
+	 */
+	public function get( $args = array() ) {
+		if ( ! $this->is_cacheable() ) {
+			return false;
+		}
+		if ( file_exists( $this->cache_filename( $args ) ) ) {
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents, WordPress.Security.EscapeOutput.OutputNotEscaped
+			echo file_get_contents( $this->cache_filename( $args ) ) . '<!-- cached -->';
+			die();
+		}
+		return false;
+	}
+
+	/*
+	 * Creates the cache directory if it doesn't exist.
+	 *
+	 * @param string $path - The path to the cache directory to create.
+	 */
+	private function create_cache_directory( $path ) {
+		if ( ! is_dir( $path ) ) {
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.dir_mkdir_dirname, WordPress.WP.AlternativeFunctions.file_system_operations_mkdir
+			return mkdir( $path, 0755, true );
+		}
+
+		return true;
+	}
+
+	/*
+	 * Saves the output buffer to the cache file for the given request, or current request.
+	 * Then outputs the buffer to the browser.
+	 *
+	 * @param string $buffer - The output buffer to save to the cache file.
+	 * @return bool|WP_Error - WP_Error if page was not cacheable.
+	 */
+	public function set( $buffer ) {
+		if ( ! $this->is_cacheable() ) {
+			return new \WP_Error( 'Page is not cacheable' );
+		}
+
+		if ( strlen( $buffer ) === 0 ) {
+			return new \WP_Error( 'Empty buffer' );
+		}
+
+		$cache_filename = $this->cache_filename();
+		if ( ! $this->create_cache_directory( dirname( $cache_filename ) ) ) {
+			return new \WP_Error( 'Could not create cache directory' );
+		}
+
+		$tmp_filename = $cache_filename . uniqid( wp_rand(), true ) . '.tmp';
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_operations_file_put_contents
+		if ( ! file_put_contents( $tmp_filename, $buffer ) ) {
```

Comment:
> This function may return Boolean false, but may also return a non-Boolean value which evaluates to false. Please read the section on [Booleans](https://www.php.net/manual/en/language.types.boolean.php) for more information. Use [the === operator](https://www.php.net/manual/en/language.operators.comparison.php) for testing the return value of this function.

Just to be safe, let's do a value/type check here.

Reply:
Thanks. I use file_put_contents in the is_writeable check too, so I fixed the comparison there too.

Reply:
Nice catch @dilirity